numpy/core/einsumfunc.py

```diff
@@ -0,0 +1,597 @@
+from numpy.core.multiarray import einsum as c_einsum
+from numpy.core.numeric import asarray, asanyarray, result_type
+
+__all__ = ['einsum']
+
+
+einsum_symbols = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+einsum_symbols_set = set(einsum_symbols)
+
+def _compute_size_by_dict(indices, idx_dict):
```

Comment:
Explicit doctest style examples for these helper functions would make it more obvious how they work